From 3bceb316793752a27e77e9a55bfe45f8bb40271b Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Mon, 2 Dec 2013 16:17:51 +0800
Subject: [PATCH 246/259] ENGR00290321 change of AvcHeaderType and bug fixes

VPU lib v5.4.21

- Removed 2 items from enum AvcHeaderType to support FW v3.1.1
App (unit test) must be rebuilt

- Bug 1: MJPEG: vpu_DecBitBufferFlush doesn't free buffer space
frameOffset was not cleared

- Bug 2: Encoder mvInfo report not correct
misused ADDR_MB_BASE_OFFSET

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 vpu/vpu_lib.c | 5 +++--
 vpu/vpu_lib.h | 4 +---
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 95af9ca..aed65da 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -2005,7 +2005,7 @@ RetCode vpu_EncGetOutputInfo(EncHandle handle, EncOutputInfo * info)
 		if (info->mvInfo.addr && info->mvInfo.size) {
 			size = (info->mvInfo.size + 7) / 8 * 8;
 			dst_addr = (Uint8 *)info->mvInfo.addr;
-			src_addr = (Uint8 *)(virt_addr + ADDR_MB_BASE_OFFSET);
+			src_addr = (Uint8 *)(virt_addr + ADDR_MV_BASE_OFFSET);
 			CopyBufferData(dst_addr, src_addr, size);
 		}
 	}
@@ -4693,7 +4693,8 @@ RetCode vpu_DecBitBufferFlush(DecHandle handle)
 	if (!is_mx6x_mjpg_codec(pCodecInst->codecMode)) {
 		BitIssueCommand(pCodecInst, DEC_BUF_FLUSH);
 		while (VpuReadReg(BIT_BUSY_FLAG)) ;
-	}
+	} else
+		pDecInfo->jpgInfo.frameOffset = 0;
 
 	pDecInfo->streamWrPtr = pDecInfo->streamBufStartAddr;
 
diff --git a/vpu/vpu_lib.h b/vpu/vpu_lib.h
index 769c748..5ef1bca 100644
--- a/vpu/vpu_lib.h
+++ b/vpu/vpu_lib.h
@@ -615,8 +615,6 @@ typedef enum {
 typedef enum {
 	SPS_RBSP,
 	PPS_RBSP,
-	END_SEQ_RBSP,
-	END_STREAM_RBSP,
 	SPS_RBSP_MVC,
 	PPS_RBSP_MVC
 } AvcHeaderType;
@@ -685,7 +683,7 @@ static const SocInfo soc_info[] = {
  * v4.2.2 [2008.09.03] support encoder on MX51
  * v4.0.2 [2008.08.21] add the IOClkGateSet() for power saving.
  */
-#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 4, 20)
+#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 4, 21)
 
 extern unsigned int system_rev;
 
-- 
1.8.0

